apps/chat/bots.py

```diff
@@ -103,19 +104,29 @@ def _call_predict(self, input_str, save_input_to_history=True, attachments: list
             tag, chain = self._get_child_chain(input_str, attachments)
         else:
             tag, chain = None, self.chain
+
         result = chain.invoke(
             input_str,
             config={
                 "configurable": {
                     "save_input_to_history": save_input_to_history,
+                    "save_output_to_history": self.terminal_chain is None,
                     "experiment_tag": tag,
                 }
             },
             attachments=attachments,
         )
 
         if self.terminal_chain:
-            result = self.terminal_chain.invoke(result.output)
+            result = self.terminal_chain.invoke(
+                result.output,
+                config={
+                    "configurable": {
+                        "save_input_to_history": False,
+                        "experiment_tag": self.terminal_chain_tag,
```

Comment:
Should we still not tag it with the 'processor' chain tag since that's the one that actually responded right? If we end up with multiple terminal bots we could pass a list of tags here, one for the processor and one for the terminal.

Reply:
Ah yes you are correct